### PR TITLE
Adds settings for distributing projects in Sonatype

### PR DESCRIPTION
**Pull request by rafaparadela**
This PR adds setting for distribution of projects on Sonatype.

@dialelo @raulraja Could you take a look please? Thank you.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -40,17 +40,16 @@ lazy val server = (project in file("server"))
...
       filters,
       jdbc,
       evolutions,
       cache,
     ws,
-      "org.scalaexercises" %% "runtime" % "0.0.0-SNAPSHOT" changing(),
-      "org.scalaexercises" %% "content-stdlib" % "0.0.0-SNAPSHOT",
-      "org.scalaexercises" %% "content-cats" % "0.0.0-SNAPSHOT",
-      "org.scalaexercises" %% "content-shapeless" % "0.0.0-SNAPSHOT",
+      "org.scala-exercises" %% "runtime" % "0.0.1" changing(),
```
Comment (raulraja) [inline build.sbt]: These shall remain snapshots until we have a final version because we want up to date snapshots of master.

File: build.sbt [hunk 2]
```diff
@@ -217,6 +220,45 @@ lazy val `sbt-exercise` = (project in file("sbt-exercise"))
...
 
+// Distribution
+
+lazy val gpgFolder = sys.env.getOrElse("SE_GPG_FOLDER", ".")
+
+lazy val publishSettings = Seq(
+  organizationName := "Scala Exercises",
+  organizationHomepage := Some(new URL("http://scala-exercises.org")),
+  startYear := Some(2016),
+  description := "Scala Exercises: The path to enlightenment",
```
Comment (raulraja) [inline build.sbt]: We need a better tagline. Maureen may be of help here.

File: build.sbt [hunk 3]
```diff
@@ -217,6 +220,45 @@ lazy val `sbt-exercise` = (project in file("sbt-exercise"))
...
+  homepage := Some(url("http://scala-exercises.org")),
+  pgpPassphrase := Some(sys.env.getOrElse("SE_GPG_PASSPHRASE", "").toCharArray),
+  pgpPublicRing := file(s"$gpgFolder/pubring.gpg"),
+  pgpSecretRing := file(s"$gpgFolder/secring.gpg"),
+  credentials += Credentials("Sonatype Nexus Repository Manager",  "oss.sonatype.org",  sys.env.getOrElse("PUBLISH_USERNAME", ""),  sys.env.getOrElse("PUBLISH_PASSWORD", "")),
+  scmInfo := Some(ScmInfo(url("https://github.com/scala-exercises/site"), "https://github.com/scala-exercises/site.git")),
+  licenses := Seq("Apache License, Version 2.0" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt")),
+  publishMavenStyle := true,
+  publishArtifact in Test := false,
+  pomIncludeRepository := Function.const(false),
```
Comment (raulraja) [inline build.sbt]: What is Function.const?

Comment (rafaparadela) [inline build.sbt]: Given that  `pomIncludeRepository` expect a function that returns a boolean, I used  `Function.const` which represents the constant function:

```scala
def const[T, U](x: T)(y: U): T = x
```

Actually we can express it thus:

```scala
pomIncludeRepository := { _ => false }
```

File: build.sbt (hunk 2 above)
Comment (rafaparadela) [inline build.sbt]: Makes sense

File: build.sbt (hunk 1 above)
Comment (rafaparadela) [inline build.sbt]: 👍

File: build.sbt (hunk 3 above)
Comment (raulraja) [inline build.sbt]: No,  it's great, just I have never seen that  before.

**Comment by raulraja** [comment]
LGTM!  tagline can be separately addressed.

**Pull request merged**